modules/compression/include/ips_jpg_at_model.cpp

```diff
@@ -0,0 +1,226 @@
+  
+  //compression signals
+  sc_out<sc_int<32> > output_size_signal;
+  
+  //-----------Internal variables-------------------
+  //const int Block_rows = 8;
+  //const int Block_cols = 8;
+  double* image;
+  int image_rows = 480;
+  int image_cols = 640;
+  signed char EOB = 127; // end of block
+  
+  int quantificator[8][8] = { // quantization table
+  {16,11,10,16,24,40,51,61},
+  {12,12,14,19,26,58,60,55},
+  {14,13,16,24,40,57,69,56},
+  {14,17,22,29,51,87,80,62},
+  {18,22,37,56,68,109,103,77},
+  {24,35,55,64,81,104,113,92},
+  {49,64,78,87,103,121,120,101},
+  {72,92,95,98,112,100,103,99}};
+
+  int zigzag_index[64]={ // zigzag table
+  0,1,5,6,14,15,27,28,
+  2,4,7,13,16,26,29,42,
+  3,8,12,17,25,30,41,43,
+  9,11,18,24,31,40,44,53,
+  10,19,23,32,39,45,52,54,
+  20,22,33,38,46,51,55,60,
+  21,34,37,47,50,56,59,61,
+  35,36,48,49,57,58,62,63};
+
+  sc_event starter_event;
+  sc_event input_event;
+  sc_event output_event;
+  sc_event compression_event;
+
+  // Constructor for compressor
+  SC_HAS_PROCESS(jpg_output);
+    jpg_output(sc_module_name jpg_compressor): sc_module(jpg_compressor){
+      image = new double[image_rows*image_cols];
+      //initialize the image matrix to avoid nan
+      for(int i=0; i<(image_rows*image_cols);i++){
+		image[i]=0;
+	  }
+	  SC_THREAD(starter_operation);
+	  SC_THREAD(input_operation);
+	  SC_THREAD(output_operation);
+	  SC_THREAD(compression_operation);
+  } // End of Constructor
+
+  //------------Code Starts Here-------------------------
+  void Starter() {
+	starter_event.notify(4, SC_NS);
+  }
+
+  void starter_operation(){
+	  while(true) {
+		wait(starter_event);
+        int im_rows = row_signal.read();
+	    if(im_rows%Block_rows==0) {image_rows=im_rows;}
+        else {image_rows=(im_rows/Block_rows+1)*Block_rows;}
+		wait(4, SC_NS);
+		int im_cols = col_signal.read();
+        if(im_cols%Block_cols==0) {image_cols=im_cols;}
+        else {image_cols=(im_cols/Block_cols+1)*Block_cols;}
+	  }
+  }
+
+  void InputPixel() {
+	input_event.notify(8, SC_NS);
+  }
+
+  void input_operation(){
+	  while(true) {
+		wait(input_event);
+		double* i_row = &image[row_signal.read() * image_cols];
+		i_row[col_signal.read()] = double(PixelValue_signal.read());
+	  }
+  }
+
+  //void OutputPixel(int *Pixel, int row, int col) {
+  //  double* i_row = &image[row * image_cols];
+  //  *Pixel = int(i_row[col]);
+  //}
+
+  void OutputByte() {
+	output_event.notify(8, SC_NS);
+  }
+
+  void output_operation(){
+	  while(true) {
+		wait(output_event);
+        Element_signal = image[index_signal.read()];
+	  }
+  }
+
+  void JPEG_compression() {
+	  compression_event.notify(100, SC_NS);
+  }
+
+  void compression_operation() {
+    while(true) {
+		wait(compression_event);
+        int output_size = 0;
+		//Level shift
+		for(int i=0; i<(image_rows*image_cols);i++){
+			image[i]=image[i]-128;
+		}
+		wait(100, SC_NS);
+		int Number_of_blocks = image_rows*image_cols/(Block_rows*Block_cols);
```

Comment:
Use consistent notation for methods and variable names like camelCase, snake_case, or PascalCase.